lib/evaluate/intrinsics.cc

```diff
@@ -1451,6 +1463,14 @@ static bool ApplySpecificChecks(
       messages.Say(
           "Arguments of ASSOCIATED() must be a POINTER and an optional valid target"_err_en_US);
     }
+  } else if (name == "loc") {
+    if (const auto &arg{call.arguments[0]}) {
+      ok = GetLastSymbol(arg->UnwrapExpr()) != nullptr;
+    }
+    if (!ok) {
+      messages.Say(
+          "Argument of LOC() must be an object or procedure"_err_en_US);
```

Comment:
What if the argument is something that has a symbol but is not an object or procedure, like a derived type or module?

Reply:
`GetLastSymbol()` should return a null pointer in those cases.

Reply:
More: those objects can't be represented in the expression data structure, so `GetLastSymbol()` would never be able to see them.

Reply:
OK, thanks